packages/contracts/contracts/BraindeadFeed.sol

```diff
@@ -0,0 +1,262 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.17;
+
+import "./Interfaces/IPriceFeed.sol";
+import "./Interfaces/IOracleCaller.sol";
+import "./Dependencies/AuthNoOwner.sol";
+
+
+/*
+ * PriceFeed for mainnet deployment, it connects to two Chainlink's live feeds, ETH:BTC and
+ * stETH:ETH, which are used to aggregate the price feed of stETH:BTC in conjuction.
+ * It also allows for a fallback oracle to intervene in case that the primary Chainlink oracle fails.
+ *
+ * The PriceFeed uses Chainlink as primary oracle and allows for an optional fallback source. It contains logic for
+ * switching oracles based on oracle failures, timeouts, and conditions for returning to the primary
+ * Chainlink oracle. In addition, it contains the mechanism to add or remove the fallback oracle through governance.
+ */
+contract BraindeadFeed is IPriceFeed, AuthNoOwner {
+    string public constant NAME = "PriceFeed";
+
+    // The last good price seen from an oracle by Liquity
+    uint256 public lastGoodPrice;
+
+    address public primaryOracle;
+    address public secondaryOracle;
+
+    uint256 INVALID_PRICE = 0;
+
+    // NOTE: Could still use Status to signal current FSM
+
+    // --- Dependency setters ---
+
+    /// @notice Sets the addresses of the contracts and initializes the system
+    constructor(
+        address _primaryOracle,
+        address _secondaryOracle
+    ) {
+        uint256 firstPrice = IOracleCaller(_primaryOracle).getLatestPrice();
+        require(firstPrice != 0, "Primary Oracle Must Work");
+
+        _storePrice(firstPrice);
+
+        primaryOracle = _primaryOracle;
+
+        // If secondaryOracle is known at deployment let's add it
+        if(_secondaryOracle != address(0)) {
+            uint256 secondaryOraclePrice = IOracleCaller(_secondaryOracle).getLatestPrice();
+
+            if(secondaryOraclePrice != 0) {
+                secondaryOracle = _secondaryOracle;
+            }
+        }
+    }
+
+    function setPrimaryOracle(address _newPrimary) external requiresAuth {
+        uint256 currentPrice = IOracleCaller(_newPrimary).getLatestPrice();
```

Comment:
do we need to use `tinfoilCall()` to `_newPrimary` here instead calling directly?

Reply:
No because if the call reverts normally, the oracle should not be added